src/app/static/src/js/components/vuex/runReport/gitSelections.vue

```diff
@@ -3,35 +3,112 @@
+                        {{ commit.id }} ({{ commit.date_time }})
+                    </option>
+                </select>
+            </div>
+        </div>
     </div>
 </template>
 
 <script lang="ts">
     import Vue from "vue";
     import {mapState} from "vuex";
-    import {EmptyObject, RunReportMetadataDependency} from "../../../utils/types";
+    import {RunReportRootState} from "../../../store/runReport/store";
+    import {EmptyObject, GitCommit, RunReportMetadataDependency} from "../../../utils/types";
+    import {mapActionByName} from "../../utils";
+    import {GitAction} from "../../../store/git/actions";
     import {GitState} from "../../../store/git/git";
     import {namespace} from "../../../store/runReport/store";
 
+    interface Data {
+        newSelectedBranch: string
+        newSelectedCommitId: string
+    }
+
     interface Computed {
         metadata: RunReportMetadataDependency
         gitBranches: string[]
+        gitCommits: GitCommit[]
+        selectedCommitId: string
+        showCommits: boolean
+    }
+
+    interface Methods {
+        selectBranch: (branch: string) => void
+        selectCommit: (branch: string) => void
+        preSelectBranch: () => void
+        preSelectCommit: () => void
     }
 
-    export default Vue.extend<EmptyObject, EmptyObject, Computed, EmptyObject>({
+    export default Vue.extend<Data, Methods, Computed, EmptyObject>({
         name: "GitSelections",
+        data(){
+            return {
+                newSelectedBranch: "",
+                newSelectedCommitId: ""
+            }
+        },
         computed: {
-            ...mapState(namespace.git, {
-                metadata: (state: GitState) => state.metadata,
-                gitBranches: (state: GitState) => state.branches
-            })
+            ...mapState({
+                metadata: (state: RunReportRootState) => state.git.metadata,
+                gitBranches: (state: RunReportRootState) => state.git.branches,
+                selectedBranch: (state: RunReportRootState) => state.git.selectedBranch,
+                gitCommits: (state: RunReportRootState) => state.git.commits,
+                selectedCommitId: (state: RunReportRootState) => state.git.selectedCommit
+            }),
+            showCommits() {
+                return !!this.gitCommits?.length;
+            }
+        },
+        methods: {
+            selectBranch: mapActionByName("git", GitAction.SelectBranch),
+            selectCommit: mapActionByName("git", GitAction.SelectCommit),
+            preSelectBranch() {
+                const selectedBranch = this.selectedBranch
+                if (selectedBranch && this.gitBranches.some(branch => branch === selectedBranch)) {
+                    this.newSelectedBranch = selectedBranch;
+                } else if (this.gitBranches.length) {
+                    this.newSelectedBranch = this.gitBranches[0]
+                }
```

Comment:
if (this.gitBranches.length) {
                    const selectedBranchExists = this.gitBranches.some(branch => branch === this.selectedBranch)
                    this.newSelectedBranch = selectedBranchExists ? this.selectedBranch : this.gitBranches[0]
                }